brainscore_core/plugin_management/generate_benchmark_metadata.py

```diff
@@ -0,0 +1,278 @@
+                  file=sys.stderr)
+            yaml_path = self.process_single_benchmark(benchmark_name)
+            if yaml_path:
+                yaml_paths.add(yaml_path)
+        return list(yaml_paths)
+
+    def find_registered_benchmarks(self, root_folder: str) -> List[str]:
+        """
+        - Finds all registered benchmarks inside `__init__.py` files within a given root directory.
+
+        :param root_folder: str, the root directory to search for benchmark registrations.
+        :return: List[str], a list of benchmark names found in `benchmark_registry` assignments.
+
+        Notes:
+        - Recursively searches for `__init__.py` files in the specified directory.
+        - Extracts benchmark names assigned to `benchmark_registry[...]` using regex.
+        - Logs an error message if any `__init__.py` file cannot be read.
+        """
+
+        registered_benchmarks = []
+        init_file_path = os.path.join(root_folder, "__init__.py")
+
+        # Ensure that `root_folder` is a benchmark directory (must contain `__init__.py`)
+        if not os.path.isfile(init_file_path):
+            print(f"ERROR: {root_folder} does not contain an `__init__.py` file.", file=sys.stderr)
+            return []
+        try:
+            with open(init_file_path, "r", encoding="utf-8") as file:
+                content = file.read()
+            matches = re.findall(r'benchmark_registry\[\s*["\'](.*?)["\']\s*\]\s*=', content, re.DOTALL)
+            if matches:
+                registered_benchmarks.extend(matches)
+        except Exception as e:
+            print(f"ERROR: Could not read {init_file_path}: {e}", file=sys.stderr)
+        return registered_benchmarks
+
+    def load_benchmark(self, identifier: str) -> Optional[object]:
+        """
+        - Loads a benchmark using `brainscore_vision` and returns the benchmark instance.
+
+        :param identifier: str, the unique name of the benchmark to load.
+        :return: Optional[object], the benchmark instance if successfully loaded, otherwise None.
+
+        Notes:
+        - Uses `import_plugin` to dynamically load the benchmark from `brainscore_vision.benchmark`.
+        - Retrieves the benchmark instance from `benchmark_registry` using the given identifier.
+        - Returns `None` if an error occurs during benchmark loading.
+        - Prints an error message if the benchmark fails to load.
+        """
+
+        try:
+            import_plugin('brainscore_vision', 'benchmarks', identifier)
+            from brainscore_vision import benchmark_registry
+            benchmark_instance = benchmark_registry[identifier]()
+            return benchmark_instance
+        except Exception as e:
+            error_message = f"ERROR: Failed to load benchmark '{identifier}': {e}"
+            print(error_message, file=sys.stderr)
+            return None, None
+
+    def create_stimuli_metadata(self, plugin, plugin_dir_name):
```

Comment:
this would be good to separate into the domains. Provide the general scaffold in core, implement details in the domains. See here for an example: https://github.com/brain-score/core/blob/44a7b7c5f10e132318c9c68e0f3d17a3d4bbc156/brainscore_core/submission/endpoints.py#L94-L106